pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -438,6 +438,22 @@ func (s *Reconciler) Delete(ctx context.Context) error {
 		return fmt.Errorf("failed to delete OS disk: %w", err)
 	}
 
+	for _, dd := range s.scope.MachineConfig.DataDisks {
+		dataDiskSpec := &disks.Spec{
+			Name: azure.GenerateDataDiskName(s.scope.Machine.Name, dd.NameSuffix),
+		}
+
+		err = s.disksSvc.Delete(ctx, dataDiskSpec)
+		if err != nil {
```

Comment:
Nit: please don't shadow the declaration of a previous error when the only return variable is the error, inline this insetad
```suggestion
		if err := s.disksSvc.Delete(ctx, dataDiskSpec); err != nil {
```